pkg/handlers/v1/cloud_fetch.go

```diff
@@ -42,9 +56,42 @@ type CloudAssetFetchByHostnameParameters struct {
 // CloudAssetFetchAllByTimestampParameters represents the incoming payload for bulk fetching cloud assets for point in time with optional pagination
 type CloudAssetFetchAllByTimestampParameters struct {
 	Timestamp string `json:"time"`
-	// we use the pointer type to detect if the value was not present in input as otherwise the int variable would be 0, which is a valid input
-	Count  *uint `json:"count"`
-	Offset *uint `json:"offset"`
+	Count     uint   `json:"count"`
+	Offset    uint   `json:"offset"`
+	Type      string `json:"type"`
+}
+
+func (p *CloudAssetFetchAllByTimestampParameters) toNextPageToken() (string, error) {
```

Comment:
nit: Personally, I prefer not having behaviors/methods attached to data containers. My recommendation is generally that 1) interfaces define behaviors but do not contain data, 2) structs contain data but do not implement behaviors, and 3) a struct may implement and interface but then it must not contain data. The phrasing of "contains data" here means that structs attributes are exported and we expect people to use them directly. The goal in that philosophy is to ensure that we don't fall into traditional class patterns where we're trying to model state and ways of manipulating that state. It's not perfect and there are no "hard lines" in practice but I find it useful.

For example, instead of `func(p *CloudFetchAllbyTimestampParameters) toNextPageToken()` I'd suggest `func NextPageToken(p CloudFetchAllByTimestampParameters)`.  Practically speaking, these are both the same. The receiver pointer notation like `func(*Something)` is intended to replace old C-style struct methods that eventually became class/instance methods. This is really more of a philosophical change where receiver functions operate on the state of the instance they are attached to and the non-receiver style is a pure function that operates on a given input.

Again, they aren't that different in most cases. I just find it more comfortable to avoid any accidental state that _might_ show up in class-style methods.

Reply:
Excellent explanation, esp the "receiver functions operate on the state of the instance they are attached to and the non-receiver style is a pure function that operates on a given input" part.